src/components/Card/Card.test.tsx

```diff
@@ -1,37 +1,98 @@
+
 const DEFAULT_PROPS: CardProps = {
-  title: 'Title Text',
-  value: '123',
-  bottomText: 'Bottom Text'
+  title: '',
+  value: '',
+  bottomText: undefined
 };
 
+const MOCK_SKELETON_ID = 'mock-skeleton';
+
 jest.mock('../Skeleton', () => ({
-  Skeleton: () => <span>Skeleton</span>
+  Skeleton: () => <div data-testid={MOCK_SKELETON_ID} />
 }));
 
-const renderComponent = (props?: CardProps) => render(<Card {...props} />);
+describe('<Card />', () => {
+  test('should only render text in the correct order - title, value, bottom text', () => {
+    const { container } = render(<Card {...DEFAULT_PROPS} title={'1,'} value={'2,'} bottomText={'3'} />);
+    expect(container).toHaveTextContent('1,2,3');
+  });
 
-test('should work with synchronous data', () => {
-  const { getByText } = renderComponent(DEFAULT_PROPS);
-  expect(getByText('Title Text')).toBeInTheDocument();
-  expect(getByText('123')).toBeInTheDocument();
-  expect(getByText('Bottom Text')).toBeInTheDocument();
-});
+  describe('semantic tags', () => {
+    test('should render title as <label>', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      const label = container.getElementsByTagName('label');
+      expect(label.length).toBe(1);
+      expect(label[0]).toHaveTextContent('mock title');
+    });
+  });
 
-test('should show skeleton when loading asynchronous data', () => {
-  const { getByText } = renderComponent({ ...DEFAULT_PROPS, value: { isLoading: true } });
-  expect(getByText('Skeleton')).toBeInTheDocument();
-});
+  describe('class names', () => {
+    test('should apply Title class to only the title element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      const title = container.getElementsByClassName(styles.Title);
+      expect(title.length).toBe(1);
+      expect(title[0]).toHaveTextContent('mock title');
+    });
+
+    test('should only apply Value class to only the value element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} value={'mock value'} />);
+      const value = container.getElementsByClassName(styles.Value);
+      expect(value.length).toBe(1);
+      expect(value[0]).toHaveTextContent('mock value');
+    });
+
+    test('should only apply BottomText class to the bottom text element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} bottomText={'mock bottom text'} />);
+      const bottomText = container.getElementsByClassName(styles.BottomText);
+      expect(bottomText.length).toBe(1);
+      expect(bottomText[0]).toHaveTextContent('mock bottom text');
+    });
+  });
+
+  describe('when text props are synchronous', () => {
+    test('should render title', () => {
+      render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      expect(screen.getByText('mock title')).toBeInTheDocument();
+    });
+
+    test('should render value in the correct element (<b>)', () => {
```

Comment:
If the intention is to check value is rendered in/with the correct element ```<b>```, should we use something like:
```
const value = container.getElementsByTagName('<b>');
expect(value).toHaveTextContent('mock value')
```

Reply:
Oops, forgot to update the test name. Thanks for pointing this out!